src/avalanchemq/queue/durable_queue.cr

```diff
@@ -172,7 +172,13 @@ module AvalancheMQ
           ack_count = ((ack.size - sizeof(Int32)) // SP_SIZE).to_u32
           acked = Array(SegmentPosition).new(ack_count)
           loop do
-            acked << SegmentPosition.from_io ack
+            sp = SegmentPosition.from_io ack
+            if sp.zero?
+              @log.warn { "Deleting corrupt ack index" }
+              ack.delete
```

Comment:
you should only delete the remainder of the file, not the whole file. alterantlivy rewrite/compact